### PR TITLE
Fix aliasing columns in CTE

### PR DESCRIPTION
**Pull request by kasiafi**
Before this change, column aliases were assigned to leftmost
columns of the WITH-query. That was incorrect in case when
there were any hidden columns preceding any of the visible columns.
The hidden columns should be skipped when assigning aliases,
and not considered in the resulting scope.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java [hunk 1]
```diff
@@ -1351,23 +1352,22 @@ private Scope createScopeForCommonTableExpression(Table table, Optional<Scope> s
...
-                            QualifiedName.of(table.getName().getSuffix()),
-                            Optional.of(columnName.getValue()),
-                            inputField.getType(),
-                            false,
-                            inputField.getOriginTable(),
-                            inputField.getOriginColumnName(),
-                            inputField.isAliased()));
-
-                    field++;
+                Iterator<Identifier> aliases = columnNames.get().iterator();
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java:1357]: Could you add test?

Comment (kasiafi) [reply core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java:1357]: Sure, however I don't know how to do it. Is there any known table having a hidden column somewhere "in the middle"?

Comment (sopel39) [reply core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java:1357]: Could you have a synthetic table like that? You could maybe use dummy connector as: `io.trino.execution.TestCoordinatorDynamicFiltering.TestConnector`

Comment (kasiafi) [reply core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java:1357]: When the hidden column is moved to the first position in the table, this fix is covered by a preexisting test in `TestWith` class.

Comment (sopel39) [reply core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java:1357]: Why the test did not fail then?

**Review by sopel39 — APPROVED** [review]

**Comment by sopel39** [comment]
merged thanks!

**Pull request merged**